src/main/java/study/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,70 @@
+package study.racingcar.domain;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Cars implements Iterable<Car> {
+    private List<Car> cars;
+
+    public Cars() {
+        this.cars = new ArrayList<>();
+    }
+
+    public Cars(List<String> carNames) {
+        if (Objects.isNull(carNames)) {
+            throw new IllegalArgumentException("null을 차로 입력할 수 없습니다.");
+        }
+
+        this.cars = carNames.stream()
+                .map(Car::new)
+                .collect(Collectors.toList());
+    }
+
+    public void add(Car car) {
+        this.cars.add(car);
+    }
+
+    public void add(Car... cars) {
+        this.cars.addAll(Arrays.asList(cars));
+    }
+
+    public int size() {
+        return cars.size();
+    }
+
+    public int getMaxPosition() {
+        return Collections
+                .max(cars, Comparator.comparing(Car::getPosition))
+                .getPosition();
+    }
+
+    public Cars getByPosition(int position) {
+        List<Car> matchedCars = cars.stream()
+                .filter(car -> car.isPositionEqualTo(position))
+                .map(Car::clone)
+                .collect(Collectors.toList());
+
+        Cars newCars = new Cars();
+        newCars.cars = matchedCars;
+
+        return newCars;
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return new Cars(matchedCars);
```

Reply:
사실 저도 이 부분에 대해서는 많은 고민이 있었습니다.
첫번째는 `List<Car>` 타입을 다루는 부분은 `Cars` 내부로 제한하고 싶었습니다.  외부에서 `List<Car>` 데이터를 생성한 다음 `Cars`의 생성자로 넣게 하고 싶지 않았습니다. 
두번째는 저도 언급하신 부분때문에  `Cars(List<Car> cars)`  생성자를 `private`으로 만들고 싶었는데요, `Cars(List<String> carNames)` 생성자와 충돌이 나더군요 😢 
혹시 `Cars(List<String> carNames)`를 유지하면 구조상 장점보다 단점이 더 많게 될까요? @_@

추가로 `Cars()` 생성자와 `add()` 함수도  `public`으로 만들고 싶지 않았었는데, 테스트하려고 열어두었다가 여기저기 사용하게 되었습니다. 😱 (남겨주신 [리뷰](https://github.com/next-step/java-racingcar/pull/621#discussion_r399759702)를 보니 어서 접근제한자를 바꿔야겠습니다 😺 )

Reply:
생성자 대신 정적 팩토리 메서드를 사용하면 어떨까요? 정적 팩토리 메서드는 이름만 잘 지으면 반환될 객체의 특성을 쉽게 묘사할 수 있어요. 아래의 글이 도움 되면 좋겠어요.
https://www.slipp.net/wiki/pages/viewpage.action?pageId=30771479